server/auth/simple_token.go

```diff
@@ -212,6 +212,9 @@ func (t *tokenSimple) info(ctx context.Context, token string, revision uint64) (
 
 func (t *tokenSimple) assign(ctx context.Context, username string, rev uint64) (string, error) {
 	// rev isn't used in simple token, it is only used in JWT
+	if _, ok := ctx.Value(AuthenticateParamIndex{}).(uint64); !ok {
+		return "", fmt.Errorf("failed to assign")
+	}
 	index := ctx.Value(AuthenticateParamIndex{}).(uint64)
```

Comment:
A further comment on top of @serathius 's comment. I agree with the suggested change, but you also need to define the variables (including both `index` and `ok`) beforehand, and change the "=:" to "=", otherwise the following code can't read the variable `index`.

The alternative is to get all the following code included in the `else` branch. 

You can also get all the code included in `if` branch, and change the condition `!ok` to `ok`. In this case, return  `"", errors.New("failed to assign") ` (Note that errors.New is better than fmt.Errorf because there is no additional arguments) in the end of the method (outside of the if branch).  I prefer to this choice. 

@AdamKorcz  Please let me know if you need further clarification.

Reply:
Thank you both for the feedback. Updated.